proposals/route-blinding.md

```diff
@@ -0,0 +1,242 @@
+
+Even with such mitigations the sender can discover the real fees/cltv of one of the blinded
+channels. To do so she uses the correct fees and cltv for all but one channel, and for that target
+channel she tries fees/cltv until the payment succeeds. Once the payment succeeds she knows the
+approximate fees/cltv of the target channel (but since the payment succeeded, she can't continue
+probing).
+
+Maybe nodes along the blinded path could use slightly different fees/cltv than what they publicly
+advertize? Or the recipient could add some fuzzing to it to blind them more?
+
+Are those mitigations enough? Or can a clever attacker still work around them?
+
+## Tips and Tricks
+
+### Dummy hops
+
+The sender knows an upper bound on the distance between the recipient and `N(0)`. If the recipient
+is close to `N(0)`, this might not be ideal. In such cases, the recipient may add any number of
+dummy hops at the beginning of the blinded route by using `N(j) = N(r)`. The sender will not be
+able to distinguish those from normal blinded hops.
+
+Note that the recipient needs to fully validate each dummy hop to detect tampering.
+
+### Wallets and unannounced channels
+
+Route blinding is particularly useful for wallets that are connected to nodes via unannounced
+channels. Such wallets would use a single blinded hop, which effectively hides their `node_id`
+and `scid` from the sender. It obviously reveals to the blinded node that the next node is the
+final recipient, but a wallet that's not online all the time with a stable IP will never be able
+to hide that information from the nodes it connects to anyway (even with rendezvous).
+
+### Blinded trampoline route
+
+Route blinding can also be used with trampoline very easily. Instead of encrypting the
+`outgoing_channel_id`, we simply need to encrypt the `outgoing_node_id`.
+
+Each trampoline node can then decrypt the `node_id` of the next node and compute `E(i)` for the
+next trampoline node. That `E(i)` can then be sent in the outer onion payload instead of using the
+message's TLV extensions, which is even cleaner.
+
+## FAQ
+
+### Why not use rendezvous
+
+While rendezvous is more private, it's also less flexible: it doesn't support reusing the partial
+onion nor retrying with updated fees on intermediate node failure. Route blinding has different
+trade-offs, which makes it useful for slightly different use-cases than rendezvous.
+
+### Why not use HORNET
+
+HORNET requires a slow session setup before it can provide useful speedups. In cases where you
+expect to send a single message per session (which is the case for most payments), HORNET actually
+performs worse than Sphinx in latency, bandwidth and privacy.
+
+## Open Questions
+
+* Should we include feature bits in the blinded path? It's yet another probing vector so we'd need
+  to "sanitize" them to avoid reducing the node's anonymity set...
+* Should we include cltv and feerates in the enctlv?
```

Comment:
nit: `enctlv` isn't a great name, first time I parsed it as `encrypted_cltv` instead of `encrypted_type_length_value` :)

Reply:
I totally agree, we really need a better name :D

Reply:
Changed that to `encrypted_recipient_data` for now.